networkx/algorithms/chains.py

```diff
@@ -141,6 +141,10 @@ def _build_chain(G, u, v, visited):
             u, v = v, G.nodes[v]["parent"]
         yield u, v
 
+    # Check if the root is in the graph G. If not, raise NodeNotFound
+    if root is not None and root not in G:
+        raise nx.NodeNotFound("Root node %s is not in graph" % root)
```

Comment:
```suggestion
        raise nx.NodeNotFound(f"Root node {root} is not in graph")
```
Just a nit, but I think [f-strings](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals) are generally preferred over the printf-style string interpolation.